InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.Modify.cs

```diff
@@ -0,0 +1,380 @@
+{
+    public partial class InternServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnModifyWhenInternIsNullAndLogItAsync()
+        {
+            // given
+            Intern nullIntern = null;
+            var nullInternException = new NullInternException();
+
+            var expectedInternValidationException =
+                new InternValidationException(nullInternException);
+
+            // when
+            ValueTask<Intern> modifyInternTask =
+                this.internService.ModifyInternAsync(nullIntern);
+
+            InternValidationException actualInternValidationException =
+                await Assert.ThrowsAsync<InternValidationException>(
+                    modifyInternTask.AsTask);
+
+            // then
+            actualInternValidationException.Should().BeEquivalentTo(
+                expectedInternValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameValidationExceptionAs(
+                    expectedInternValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.UpdateInternAsync(It.IsAny<Intern>()),
+                    Times.Never);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();            
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public async Task ShouldThrowValidationExceptionOnModifyWhenInternIsInvalidAndLogItAsync(
+            string invalidText)
+        {
+            // given
+            var invalidIntern = new Intern
+            {
+                FirstName = invalidText
+            };
+
+            var invalidInternException = new InvalidInternException();
+
+            invalidInternException.AddData(
+                key: nameof(Intern.Id),
+                values: "Id is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.FirstName),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.LastName),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.Email),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.PhoneNumber),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.Status),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.UpdatedDate),
+                values: new String[] { "Date is required", "Date is the same as CreatedDate" });
```

Comment:
This is not correct. This message should only contain content related to this use-case